### PR TITLE
feat: cm based show/hide default_cluster

### PR DESCRIPTION
**Pull request by ritvikmahajan17**
# Description

hide/show default cluster based on env variable ``HIDE_DEFAULT_CLUSTER`` in dashboard cm

Fixes #4285

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ResourceBrowser/ResourceList/ClusterSelector.tsx [hunk 1]
```diff
@@ -18,6 +18,9 @@ interface ClusterSelectorType {
 }
 
 export default function ClusterSelector({ onChange, clusterList, clusterId }: ClusterSelectorType) {
+    if (!window._env_.SHOW_DEFAULT_CLUSTER) {
+        clusterList = clusterList.filter((item) => item.label !== 'default_cluster')
```
Comment (eshankvaish) [inline src/components/ResourceBrowser/ResourceList/ClusterSelector.tsx:22]: pls use ID

File: src/components/ClusterNodes/ClusterSelectionList.tsx [hunk 2]
```diff
@@ -121,6 +121,7 @@ export default function ClusterSelectionList({
 
     const renderClusterRow = (clusterData: ClusterDetail): JSX.Element => {
         const errorCount = clusterData.nodeErrors ? Object.keys(clusterData.nodeErrors).length : 0
+        if (!window._env_.SHOW_DEFAULT_CLUSTER && clusterData.name === 'default_cluster') return <></>
```
Comment (eshankvaish) [inline src/components/ClusterNodes/ClusterSelectionList.tsx:124]: same use ID and filter the cluster options instead

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/ClusterNodes/ClusterSelectionList.tsx [hunk 3]
```diff
@@ -35,6 +36,9 @@ export default function ClusterSelectionList({
     const [searchApplied, setSearchApplied] = useState(false)
 
     useEffect(() => {
+        if (window._env_.HIDE_DEFAULT_CLUSTER) {
+            clusterOptions = clusterOptions.filter((item) => item.id !== DEFAULT_CLUSTER_ID)
```
Comment (eshankvaish) [inline src/components/ClusterNodes/ClusterSelectionList.tsx:40]: `clusterList` is a state variable, we should not update the state variable in this way

**Review by eshankvaish — APPROVED** [review]

**Review by eshankvaish — APPROVED** [review]
@vivek-devtron to provide the final approval

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**